src/Refact.hs

```diff
@@ -11,3 +18,15 @@ toRefactSrcSpan ss = R.SrcSpan (srcSpanStartLine ss)
 
 toSS :: Annotated a => a S -> R.SrcSpan
 toSS = toRefactSrcSpan . srcInfoSpan . ann
+
+toSrcSpan' :: Hs.HasSrcSpan a => a -> R.SrcSpan
+toSrcSpan' x =
+    R.SrcSpan (Hs.srcSpanStartLine span)
+              (Hs.srcSpanStartCol span)
+              (Hs.srcSpanEndLine span)
+              (Hs.srcSpanEndCol span)
+    where
+        span :: Hs.RealSrcSpan
+        span = case Hs.getLoc x of
+                 Hs.RealSrcSpan s -> s
+                 _ -> error "Got a bad ghc SrcSpan! Report a bug to hlint please!"
```

Comment:
Is this error fine? Not sure how we can handle a non-"real" src span here.

Reply:
I think a fake 0 0 0 0 span is the only viable option - I'd much rather not crash here, since if we get here, odds are it's a GHC bug we inherited, or someone has preprocessed their syntax tree, and a report to HLint isn't going to achieve much.